src/ui/public/courier/fetch/strategy/__tests__/search.js

```diff
@@ -32,40 +42,72 @@ describe('ui/courier/fetch/strategy/search', () => {
 
   describe('#reqsFetchParamsToBody()', () => {
     it('filters out any body properties that begin with $', () => {
-      let value;
-      search.reqsFetchParamsToBody(reqsFetchParams).then(val => value = val);
-      $rootScope.$apply();
-      expect(_.includes(value, 'foo')).to.be(true);
-      expect(_.includes(value, '$foo')).to.be(false);
+      return search.reqsFetchParamsToBody(reqsFetchParams).then(value => {
+        expect(_.includes(value, 'foo')).to.be(true);
+        expect(_.includes(value, '$foo')).to.be(false);
+      });
     });
 
     context('when indexList is not empty', () => {
       it('includes the index', () => {
-        let value;
-        search.reqsFetchParamsToBody(reqsFetchParams).then(val => value = val);
-        $rootScope.$apply();
-        expect(_.includes(value, '"index":["logstash-123"]')).to.be(true);
+        return search.reqsFetchParamsToBody(reqsFetchParams).then(value => {
+          expect(_.includes(value, '"index":["logstash-123"]')).to.be(true);
+        });
       });
     });
 
     context('when indexList is empty', () => {
-      beforeEach(() => reqsFetchParams[0].index = []);
+      beforeEach(() => {
+        reqsFetchParams.forEach(request =>  request.index = []);
+      });
+      const emptyMustNotQuery = JSON.stringify({
+        query: {
+          bool: {
+            must_not: [
+              { match_all: {} }
+            ]
+          }
+        }
+      });
 
       it('queries the kibana index (.kibana) with a must_not match_all boolean', () => {
-        const query = JSON.stringify({
-          query: {
-            bool: {
-              must_not: [
-                { match_all: {} }
-              ]
-            }
+        return search.reqsFetchParamsToBody(reqsFetchParams).then(value => {
+          expect(_.includes(value, '"index":[".kibana"]')).to.be(true);
+          expect(_.includes(value, emptyMustNotQuery)).to.be(true);
+        });
+      });
+
+      it('queries .kibana when using an empty IndexPattern and a time range with no data', () => {
```

Comment:
This test feels like it should be in its own `context` since its not using the `reqFetchParams` set up by the current `context` in the `beforeEach`. Instead of `when indexList is empty` it could be something like `when indexList is an IndexPattern`.